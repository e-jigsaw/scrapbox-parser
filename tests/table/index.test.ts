/* eslint-disable no-tabs, no-irregular-whitespace */

describe("Table", () => {
  it("Simple table", () => {
    expect(`table:hello
${"\t"}1${"\t"}2${"\t"}3
${"\t"}1 ${"\t"}2 ${"\t"}3
${"\t"}------${"\t"}------${"\t"}------
${"\t"}a${"\t"}b${"\t"}c`).toMatchSnapshotWhenParsing({ hasTitle: false });
  });

  it("Bulleted table", () => {
    expect(` table:bulleted
 ${"\t"}1${"\t"}2${"\t"}3
 ${"\t"}1 ${"\t"}2 ${"\t"}3
 ${"\t"}------${"\t"}------${"\t"}------
 ${"\t"}a${"\t"}b${"\t"}c`).toMatchSnapshotWhenParsing({ hasTitle: false });
  });

  it("Table with empty cells", () => {
    expect(`table:${" "}
${"\t"} ${"\t"}ã€€${"\t"}${"  "}
${"\t"}${"\t"}${"\t"}`).toMatchSnapshotWhenParsing({ hasTitle: false });
  });

  it("Staggered table", () => {
    expect(`table:Staggered
${"\t"}1${"\t"}2${"\t"}3${"\t"}4
${"\t"}1${"\t"}2${"\t"}3
${"\t"}1
${"\t"}1${"\t"}2
${"\t"}`).toMatchSnapshotWhenParsing({ hasTitle: false });
  });

  it("Consecutive table", () => {
    expect(`table:hello
${"\t"}1${"\t"}2${"\t"}3
${"\t"}1 ${"\t"}2 ${"\t"}3
${"\t"}------${"\t"}------${"\t"}------
${"\t"}a${"\t"}b${"\t"}c
table:hello
${"\t"}1${"\t"}2${"\t"}3
${"\t"}1 ${"\t"}2 ${"\t"}3
${"\t"}------${"\t"}------${"\t"}------
${"\t"}a${"\t"}b${"\t"}c`).toMatchSnapshotWhenParsing({ hasTitle: false });
  });

  it("Node in table cells", () => {
    expect(`table:node in table cells
${"\t"}#hashtag
${"\t"}[* deco]
${"\t"}[ ]
${"\t"}\`code\`
${"\t"}https://external.com
${"\t"}[https://external.com]
${"\t"}[left https://external.com]
${"\t"}[https://external.com right]
${"\t"}[$ x]
${"\t"}[N35.6812362,E139.7649361]
${"\t"}#hashTag
${"\t"}? helpfeel
${"\t"}$ commandLine
${"\t"}[progfay.icon]
${"\t"}[https://image.com/image.png]
${"\t"}[link]
${"\t"}plain
${"\t"}> quote
${"\t"}[[progfay.icon]]
${"\t"}[[https://image.com/image.png]]
${"\t"}[[strong]]`).toMatchSnapshotWhenParsing({
      hasTitle: false,
    });
  });
});
