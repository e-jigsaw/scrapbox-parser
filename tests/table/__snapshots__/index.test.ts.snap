// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Bulleted table: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "cells": Array [
      Array [
        Array [
          Object {
            "raw": "1",
            "text": "1",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "2",
            "text": "2",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "3",
            "text": "3",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "1 ",
            "text": "1 ",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "2 ",
            "text": "2 ",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "3",
            "text": "3",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "------",
            "text": "------",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "------",
            "text": "------",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "------",
            "text": "------",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "a",
            "text": "a",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "b",
            "text": "b",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "c",
            "text": "c",
            "type": "plain",
          },
        ],
      ],
    ],
    "fileName": "bulleted",
    "indent": 1,
    "type": "table",
  },
]
`;

exports[`Table Consecutive table: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "cells": Array [
      Array [
        Array [
          Object {
            "raw": "1",
            "text": "1",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "2",
            "text": "2",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "3",
            "text": "3",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "1 ",
            "text": "1 ",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "2 ",
            "text": "2 ",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "3",
            "text": "3",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "------",
            "text": "------",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "------",
            "text": "------",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "------",
            "text": "------",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "a",
            "text": "a",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "b",
            "text": "b",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "c",
            "text": "c",
            "type": "plain",
          },
        ],
      ],
    ],
    "fileName": "hello",
    "indent": 0,
    "type": "table",
  },
  Object {
    "cells": Array [
      Array [
        Array [
          Object {
            "raw": "1",
            "text": "1",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "2",
            "text": "2",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "3",
            "text": "3",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "1 ",
            "text": "1 ",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "2 ",
            "text": "2 ",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "3",
            "text": "3",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "------",
            "text": "------",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "------",
            "text": "------",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "------",
            "text": "------",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "a",
            "text": "a",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "b",
            "text": "b",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "c",
            "text": "c",
            "type": "plain",
          },
        ],
      ],
    ],
    "fileName": "hello",
    "indent": 0,
    "type": "table",
  },
]
`;

exports[`Table Simple table: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "cells": Array [
      Array [
        Array [
          Object {
            "raw": "1",
            "text": "1",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "2",
            "text": "2",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "3",
            "text": "3",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "1 ",
            "text": "1 ",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "2 ",
            "text": "2 ",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "3",
            "text": "3",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "------",
            "text": "------",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "------",
            "text": "------",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "------",
            "text": "------",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "a",
            "text": "a",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "b",
            "text": "b",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "c",
            "text": "c",
            "type": "plain",
          },
        ],
      ],
    ],
    "fileName": "hello",
    "indent": 0,
    "type": "table",
  },
]
`;

exports[`Table Staggered table: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "cells": Array [
      Array [
        Array [
          Object {
            "raw": "1",
            "text": "1",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "2",
            "text": "2",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "3",
            "text": "3",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "4",
            "text": "4",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "1",
            "text": "1",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "2",
            "text": "2",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "3",
            "text": "3",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "1",
            "text": "1",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [
          Object {
            "raw": "1",
            "text": "1",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "2",
            "text": "2",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [],
      ],
    ],
    "fileName": "Staggered",
    "indent": 0,
    "type": "table",
  },
]
`;

exports[`Table Table with empty cells: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "cells": Array [
      Array [
        Array [
          Object {
            "raw": " ",
            "text": " ",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "　",
            "text": "　",
            "type": "plain",
          },
        ],
        Array [
          Object {
            "raw": "  ",
            "text": "  ",
            "type": "plain",
          },
        ],
      ],
      Array [
        Array [],
        Array [],
        Array [],
      ],
    ],
    "fileName": " ",
    "indent": 0,
    "type": "table",
  },
]
`;

exports[`Table Table with hashTag: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "cells": Array [
      Array [
        Array [
          Object {
            "href": "tag",
            "raw": "#tag",
            "type": "hashTag",
          },
        ],
        Array [
          Object {
            "raw": "This is",
            "text": "This is",
            "type": "plain",
          },
          Object {
            "raw": " ",
            "text": " ",
            "type": "plain",
          },
          Object {
            "href": "tag",
            "raw": "#tag",
            "type": "hashTag",
          },
        ],
      ],
    ],
    "fileName": "table with hashTag",
    "indent": 0,
    "type": "table",
  },
]
`;

exports[`Table Table with link: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "cells": Array [
      Array [
        Array [
          Object {
            "content": "",
            "href": "Link",
            "pathType": "relative",
            "raw": "[Link]",
            "type": "link",
          },
        ],
        Array [
          Object {
            "raw": "This is ",
            "text": "This is ",
            "type": "plain",
          },
          Object {
            "content": "",
            "href": "Link",
            "pathType": "relative",
            "raw": "[Link]",
            "type": "link",
          },
        ],
      ],
    ],
    "fileName": "table with link",
    "indent": 0,
    "type": "table",
  },
]
`;
